kraken/prometheus/client.py

```diff
@@ -27,4 +30,53 @@ def alerts(prom_cli: KrknPrometheus, start_time, end_time, alert_profile):
 
             prom_cli.process_alert(alert,
                                    datetime.datetime.fromtimestamp(start_time),
-                                   datetime.datetime.fromtimestamp(end_time))
\ No newline at end of file
+                                   datetime.datetime.fromtimestamp(end_time))
+
+
+def critical_alerts(prom_cli: KrknPrometheus,
+                    summary: ChaosRunAlertSummary,
+                    run_id,
+                    scenario,
+                    start_time,
+                    end_time):
+    summary.scenario = scenario
+    summary.run_id = run_id
+    query = r"""ALERTS{severity="warning"}"""
+    logging.info("Checking for critical alerts firing post chaos")
+
+    during_critical_alerts = prom_cli.process_prom_query_in_range(
+        query,
+        start_time=datetime.datetime.fromtimestamp(start_time),
+        end_time=end_time
+
+    )
+
+    for alert in during_critical_alerts:
+        alert = ChaosRunAlert(alert["metric"]["alertname"], alert["metric"]["alertstate"], alert["metric"]["namespace"], alert["metric"]["severity"])
+        summary.chaos_alerts.append(alert)
+
+
+    post_critical_alerts = prom_cli.process_query(
```

Comment:
@chaitanyaenr I've the gut feeling that, as we discussed yesterday, in this way, the `during_critical_alerts` will be a subset of the `post_critical_alerts`, I'm pretty convinced that here we should use a query with range using the `end_time` of the chaos run as the start_time of the query range.

Reply:
We can take down etcd pod and compare the during vs post alerts I guess. Thoughts?

Reply:
Alert manager is taking some time to reconcile causing the during and post chaos alerts to overlap. It should be fixed if the wait_duration post chaos is long enough.